pennylane/tape/qscript.py

```diff
@@ -1195,11 +1214,13 @@ def to_openqasm(self, wires=None, rotations=True, measure_all=True, precision=No
         return qasm_str
 
     @classmethod
-    def from_queue(cls, queue, shots: Optional[Union[int, Sequence, Shots]] = None):
+    def from_queue(
+        cls: "QuantumScript", queue: qml.queuing.AnnotatedQueue, shots: Optional[ShotsLike] = None
```

Comment:
```suggestion
        cls: type, queue: qml.queuing.AnnotatedQueue, shots: Optional[ShotsLike] = None
```

Do we actually type hint `cls` and `self`?

Reply:
`self` no, but I have seen a lot instances of SO where `cls` is annotated generically (for example [here](https://stackoverflow.com/questions/46007544/python-3-type-hint-for-a-factory-method-on-a-base-class-returning-a-child-class)). I don't think our case here necessitates it though, but it might be nice to have a precedent of such a type hint for the future (?)